source/validate_barriers.cpp

```diff
@@ -81,11 +81,18 @@ spv_result_t ValidateMemoryScope(ValidationState_t& _,
   }
 
   if (spvIsVulkanEnv(_.context()->target_env)) {
-    if (value != SpvScopeDevice && value != SpvScopeWorkgroup &&
+    if (value == SpvScopeCrossDevice) {
+      return _.diag(SPV_ERROR_INVALID_DATA)
+             << spvOpcodeString(opcode)
+             << ": in Vulkan environment, Memory Scope cannot be CrossDevice";
+    }
+    if (_.context()->target_env == SPV_ENV_VULKAN_1_0 &&
```

Comment:
Normally I'd use a whitelist on the permitted scopes.  But the scope enum is already limited to a specific set, and CrossDevice is the only banned one after Vulkan 1.0.
So this works for me.  And it will naturally work well if/when Vulkan 1.2 comes out, for example.
The only concern is if another scope enum is added.  But we can deal with that remote possibility later.